### ISSUE TITLE
Reproduction in Medical Counting Dataset

### ISSUE THREAD

**Issue by chenllliang** [Deep-Agent/R1-V#7]
> Hi @chenllliang ,
> 
> Here are the results on the [cell image dataset](https://www.nature.com/articles/s41592-024-02233-6). I converted the segmentation task as a counting task. 
> 
> The overall loss also looks good but accuracy reward loss is noising. 
> 
> ![Image](https://github.com/user-attachments/assets/c317e066-191f-43da-abe0-4d19ce027710)
> 
> Will run the inference and share results. The inference pipeline should be the same as the base model qwen's pipeline, right? 

 _Originally posted by @JunMa11 in [#4](https://github.com/Deep-Agent/R1-V/issues/4#issuecomment-2634179772)_

**Comment by JunMa11** [comment]
thanks for highlighting this. 

Here is the dataset in case anyone wants to try as well. 

https://drive.google.com/drive/folders/1N2DfyT0uweyvm7Nt-OmEvb9QFX6HsdF3?usp=drive_link

The training and validation sets contain 1000 and 100 diverse cell images.

**Comment by JunMa11** [comment]
Some new results:

1. Train the model with only accuracy reward:

![Image](https://github.com/user-attachments/assets/ea3d682e-79a3-415e-8b60-5969894c1148)


2. Train the model with soft accuracy reward:

![Image](https://github.com/user-attachments/assets/e4cf0ae5-ae20-4c10-98ea-8d30b6dbf53a)

Both of them are not good. I think it requires a better base model, e.g., SFT the base model first.

**Comment by 0xvincii** [comment]
@JunMa11 thanks for your experiments. I agree that we need a better base model for this task. By the way, we release the evaluation script at https://github.com/Deep-Agent/R1-V?tab=readme-ov-file#evaluation, wish it is helpful.

I have a question, why the soft accurcy_reward could be negative?

**Comment by JunMa11** [comment]
@0xvincii thanks for sharing. 

I used this reward

```
1 - abs(student_answer - ground_truth)/ground_ground_truth
```

sometimes, student_answer is far away from the ground truth (e.g. 100 vs 10). then the reward will be negative

**Comment by JaNoDiggity** [comment]
@JunMa11 Could you please explain why the loss starts from zero and gradually increases? I am confused about that. I would appreciate it if you can tell me something about that!
